### DISCUSSION TITLE
Will this be added to macOS and Linux?

### DISCUSSION THREAD

**Discussion by Zach-Tolentino-161** [ThioJoe/YT-Spammer-Purge#54] [Q&A]
**I don't have a windows computer. I have tried to make it work with macOS but every try hasn't worked.**

**Comment by NagaSpntis** [comment]
> **I don't have a windows computer. I have tried to make it work with macOS but every try hasn't worked.**

Have you tried this in a Virtual machine? Or tried using WINE?

**Comment by Zach-Tolentino-161** [comment]
I will try using wine.

**Comment by wilsontulus** [comment]
@Zach-Tolentino-161 , can you try using the **YouTubeSpammerPurge.py** script by using the latest Python 3.10.x on your computer (assuming that you've already installed the required dependencies in the requirements.txt) file? Using this method, the script should work on all Python 3.10-supported platforms. Thanks.

**Comment by KendallDoesCoding** [comment]
Try to use a vm, or run it on a online service like Google Collab for now!

**Comment by rbeer** [comment]
What exactly have you tried? Provide error messages you get!

This is python. OS shouldn't make any difference.

VM to run a python script. Don't be silly, guys! 😄

My best guess for now is that you are simply missing tk.

_**All these commands require root privileges**_

MacOS:
`brew install python3-tk`
(No guarantees on this one. Feel free to correct me. I don't run Macs.)

Debian:
`apt install python3-tk`

Fedora:
`dnf install python3-tk`

RHEL:
`yum install tkinter tk-devel`

Arch:
`I'm sure you know what you are doing, Mr. Hacker` 😝

In case you want to verify:
```
pip3 install tk
python3 -m tkinter
```
That should open a window with some text and two buttons.

**Reply by KendallDoesCoding** [reply]
Yeah, it shouldn't make a difference about which OS your using...

**Reply by TechStudent10** [reply]
@rbeer Where are all the `sudo`s? I know that for Debian you need sudo to install anything. So, Debian should be `sudo apt install python3-tk`.

**Reply by rbeer** [reply]
You are correct. Installing packages requires root privileges. My bad; I was assuming that that's known.

**Reply by TechStudent10** [reply]
@rbeer you should've edited the commands

**Reply by rbeer** [reply]
Sorry, but that's way too much work. Whether your system even has the `sudo` command highly depends on your installation. Followed by whether your user is in the `sudoers` file, which again is probably in different places, depending on your distro.

"\<your distro\> how to gain root privileges" will show you how it's done for \<your distro\>. This is an already well documented question.

This is one of the drawbacks of Linux's diversity. Often, there simply isn't one answer that fits all. Like I said in the issue, I'm glad to help out with specific questions.

**Reply by TechStudent10** [reply]
yeah I guess you're right
